packages/flutter/lib/src/material/scaffold.dart

```diff
@@ -323,6 +324,26 @@ class ScaffoldMessengerState extends State<ScaffoldMessenger> with TickerProvide
   /// ```
   /// {@end-tool}
   ScaffoldFeatureController<SnackBar, SnackBarClosedReason> showSnackBar(SnackBar snackBar) {
+    assert(() {
+      if (SchedulerBinding?.instance!.schedulerPhase.index >= SchedulerPhase.persistentCallbacks.index) {
```

Comment:
@goderbauer I thought this was correctly identifying the build phase, but it is breaking customer tests. Do I have the index wrong?

Reply:
You probably want to use `==` instead of `>=`. After the persistentCallbacks we have the postFrameCallbacks phase and I imagine some people will schedule a post frame to show a snackbar to get around the problem that you cannot show them during build (the persistentCallbacks phase).

Reply:
Ah! Thank you!

Reply:
This makes a lot of sense. Thanks @goderbauer and @Piinks

Reply:
Google tests still failing and looks like I can't see the logs.

Reply:
Noticed that there were a invalid null-check. Hope all tests passes now.🤞🏾

Reply:
Google tests still failing and somehow I can't see the logs 😐

Reply:
Hey @pedromassango sorry I missed this notification, I'm checking the logs now.